.github/workflows/ci_msbuild.yaml

```diff
@@ -0,0 +1,110 @@
+        controller: [yrc1000]
+        ros2_codename: [humble]
+
+    steps:
+    - name: Uppercase controller identifier
+      id: uppercaser
+      shell: bash
+      env:
+        TEMP_VAR: "${{ matrix.controller }}"
+      run: |
+        echo "ctrlr=${TEMP_VAR^^}" >> $GITHUB_OUTPUT
+
+    - name: Checkout MotoROS2
+      uses: actions/checkout@v3
+      with:
+        path: 'motoros2'
+
+    - name: Find MSBuild and add to PATH
+      uses: microsoft/setup-msbuild@v1.3
+
+    - name: "Download M+ libmicroros (${{ matrix.ros2_codename }} on ${{ steps.uppercaser.outputs.ctrlr }})"
+      id: download_libmicroros
+      uses: dsaltares/fetch-gh-release-asset@1.1.1
+      with:
+        repo: 'yaskawa-global/motoros2'
+        version: 'latest'
+        regex: true
+        # download the M+ libmicroros distribution for the latest release of
+        # MotoROS2 corresponding to the controller this workflow is run for
+        file: ".*_libmicroros_${{ matrix.ros2_codename }}-.*_${{ matrix.controller }}\\.zip"
+        # work-around for https://github.com/dsaltares/fetch-gh-release-asset/issues/48
+        target: "./"
+
+    - name: Setup MotoROS2 build dir
+      shell: bash
+      run: |
+        mkdir -p /c/build
+        rm -rf /c/build/*
+        # we do this to shorten the path. Windows/M+ SDK sometimes does
+        cp -r "${{ github.workspace }}"/* /c/build/
+        ls -al /c/build
+
+    - name: "Find downloaded M+ libmicroros (${{ matrix.ros2_codename }} on ${{ steps.uppercaser.outputs.ctrlr }})"
+      id: find_libmicroros
+      shell: bash
+      # bit of a kludge, but works for now.
+      # we need to do this as 'fetch-gh-release-asset' will try to rename downloaded
+      # files if we use a regex to match 'unknown' files instead of hard-coding
+      # everything.
+      run: |
+        LIBM_ZIP=$(find /c/build -type f -regextype posix-extended -regex ".*[[:digit:]]+_libmicroros_${{ matrix.ros2_codename }}.*${{ matrix.controller}}\.zip")
+        echo "libmicroros_zip=${LIBM_ZIP}" >> $GITHUB_OUTPUT
+        echo "libmicroros_zip_win=$(cygpath -w ${LIBM_ZIP})" >> $GITHUB_OUTPUT
+        echo "Found libmicroros zip: '${LIBM_ZIP}'"
+    - name: Extract M+ libmicroros distribution
+      shell: bash
+      run: |
+        unzip -q "${{ steps.find_libmicroros.outputs.libmicroros_zip }}" \
+          -d /c/build/motoros2/libmicroros_${{ matrix.controller }}_${{ matrix.ros2_codename }}
+
+    - name: Download and setup M+ SDK (mpBuilder)
+      shell: cmd
+      run: |
+        C:\msys64\usr\bin\wget.exe -q -O C:/build/mpsdk_ci.7z "${{ secrets.MPSDK_DOWNLOAD_URL }}"
+        "C:\Program Files\7-Zip\7z.exe" x C:/build/mpsdk_ci.7z -oC:/build/
+        del /q C:\build\mpsdk_ci.7z
+    - name: Add M+ SDK to PATH
+      shell: bash
+      run: echo "C:/build/mpsdk" >> $GITHUB_PATH
+
+    - name: Register mpBuilder problem matcher
+      run: echo "::add-matcher::motoros2/.github/matchers/mpbuilder.json"
```

Comment:
The problem matcher actually works, and should result in warnings and/or errors in the output of `mpBuilder` to be converted into annotations on the files contributed in a PR.

However, `mpBuilder` currently doesn't output the full path to files, which makes it impossible for GitHub to match the errors to the respective files in the repository. The best we can do right now is to have the annotations appear on the summary page of a specific run.

Reply:
It still shows the file and line number. I think we'll be fine.
(Though I can work on a future enhancement for M+SDK at some point.)

![image](https://github.com/Yaskawa-Global/motoros2/assets/4623811/a657767f-a52f-415a-9c2e-2382481a1663)

Reply:
> It still shows the file and line number.

yes, but the links in the summary only take you to the log.

With the full path, Github would be able to map it to the file in the repository which would:

 - allow you to click on the warning/error and be taken to the actual file at the line causing the issue
 - make GitHub show in-line annotations for all warnings/errors reported in the log